examples/data_ops/1151_digits.py

```diff
@@ -0,0 +1,165 @@
+"""
+Handwritten digit classification with skrub
+============================================
+
+This example demonstrates how to use skrub's Data Ops to build a
+machine learning pipeline for handwritten digit classification. We'll cover:
+
+1. Loading and exploring the UCI ML hand-written digits dataset
+2. Building a basic classification model with skrub Data Ops
+3. Performing hyperparameter tuning with multiple algorithms
+4. Comparing model performance before and after tuning
+
+This example is inspired by the official scikit-learn example:
+https://scikit-learn.org/stable/auto_examples/classification/plot_digits_classification.html
+"""
+
+# %%
+# Loading the Data
+# ================
+#
+# We use the UCI ML hand-written digits dataset provided by scikit-learn.
+# This dataset contains 8x8 pixel images of handwritten digits (0-9) that have
+# been flattened to 64-dimensional feature vectors. Our goal is to build a
+# classifier that can accurately predict the digit from the pixel values.
+
+from sklearn.datasets import load_digits
+
+digits = load_digits()
+X, y = digits.data, digits.target
+print(f"Dataset shape: {X.shape}")
+print(f"Number of classes: {len(set(y))}")
+# %%
+# Visualizing Sample Digits
+# ==========================
+#
+# Let's visualize some examples from the dataset to understand what we're
+# working with. Each image is an 8x8 grid of pixel intensities representing
+# a handwritten digit.
+
+import matplotlib.pyplot as plt
+
+fig, axes = plt.subplots(2, 5, figsize=(10, 5))
+for ax, image, label in zip(axes.flatten(), X, y):
+    image = image.reshape(8, 8)
+    ax.imshow(image, cmap=plt.cm.gray_r, interpolation="nearest")
+    ax.set_title(f"Label: {label}")
+plt.tight_layout()
+# %%
+# Building a Baseline Model
+# ==========================
+#
+# Now we'll build our first model using skrub's Data Ops framework.
+# First, we define our X and y variables as Data Ops.
+
+import skrub
+
+X = skrub.X(X)
+y = skrub.y(y)
+
+# %%
+# Starting with a Simple SVM
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# We begin with a Support Vector Machine (SVM) classifier. The skrub
+# Data Ops framework allows us to define our model and apply it to the data
+# in a declarative way using the apply method.
+
+from sklearn.svm import SVC
+
+model = SVC()
+predictions = X.skb.apply(model, y=y)
```

Comment:
here also we could insist on what distinguishes this example / what it adds wrt to the other hyperparam search examples. is it that we are using numpy arrays rather than dataframes?

maybe we could go one step further and use pytorch instead of numpy+sklearn, given that we have it installed for the example gallery anyway due to the textencoder. it's more often used for processing images and "can I use this other framework" is a question that comes up